packages/react-components/lib/tasks/types/delivery-custom.tsx

```diff
@@ -255,6 +332,36 @@ const isDeliveryPickupTaskDescriptionValid = (
   );
 };
 
+const isDoubleComposeDeliveryTaskDescriptionValid = (
+  taskDescription: DoubleComposeDeliveryTaskDescription,
+  pickupPoints: Record<string, string>,
+  dropoffPoints: Record<string, string>,
+): boolean => {
+  const firstGoToPickup = taskDescription.phases[0].activity.description.activities[0];
+  const firstPickup = taskDescription.phases[0].activity.description.activities[1];
+  const firstGoToDropoff = taskDescription.phases[1].activity.description.activities[0];
+
+  const secondGoToPickup = taskDescription.phases[3].activity.description.activities[0];
+  const secondPickup = taskDescription.phases[3].activity.description.activities[1];
+  const secondGoToDropoff = taskDescription.phases[4].activity.description.activities[0];
```

Comment:
I didn't check if the number of phases and activities is correct.

Reply:
the phase types and activity types are encoded in the interface already, if there is a mismatch, it will be a compiler error